fabric-api-lookup-api-v1/src/main/java/net/fabricmc/fabric/api/lookup/v1/custom/ApiLookupMap.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.lookup.v1.custom;
+
+import java.util.Objects;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.lookup.custom.ApiLookupMapImpl;
+
+//CHECKSTYLE.OFF: JavadocStyle - Checkstyle didn't like <A, C>, even though {@code ... } already escapes it.
+/**
+ * A a map meant to be used as the backing storage for custom {@code ApiLookup} instances,
+ * to implement a custom equivalent of {@link net.fabricmc.fabric.api.lookup.v1.block.BlockApiLookup#get BlockApiLookup#get}.
+ *
+ * <p><h3>Usage Example</h3>
+ * We will be implementing the following simplified version of an API lookup interface for item stacks
+ * to illustrate how to use {@link ApiLookupMap} and {@link ApiProviderMap}.
+ * <pre>{@code
+ * public interface ItemStackApiLookup<A, C> {
+ *     static <A, C> ItemStackApiLookup<A, C> get(Identifier lookupId, Class<A> apiClass, Class<C> contextClass) {
+ *         return ItemStackApiLookupImpl.get(lookupId, apiClass, contextClass);
+ *     }
+ *     // Find an API instance.
+ *     ＠Nullable
+ *     A find(ItemStack stack, C context);
+ *     // Expose the API for some item.
+ *     void register(ItemStackApiProvider<A, C> provider, Item item);
```

Comment:
the reason checkstyle complained here is you need to use the html entity codes for `<` and `>`

Reply:
Then it fails to render correctly as the `{@code}` tag already escapes them, I tried that first...

Reply:
It won't render in IDE but the doclet will render it

Reply:
`gradlew javadoc` was unable to render them correctly when I tried on my laptop - just disable the inspection and be done with it, there are more important things in life and javadoc escaping is weird anyway. :P

Reply:
Fun fact: the `＠` symbol in the javadoc is not a regular `@` 😆 (https://stackoverflow.com/a/46332643/13567109).